# Module 2 Lab 1: Python Warm-Up + First Push

This notebook demonstrates core Python concepts including variables, f-strings, lists, loops, functions, and dictionaries, as well as loading a CSV file with pandas.

In [11]:
name = "Mac"
hourly_wage = 60.00
hours = 28

weekly_pay = hourly_wage * hours

print(f"{name} earns ${weekly_pay:.2f} per week.")

Mac earns $1680.00 per week.


In [12]:
temps = [61, 65, 72, 58, 70, 74, 68]
warm_days = 0

for temperature in temps:
    if temperature > 65:
        warm_days = warm_days + 1

print(warm_days)

4


In [13]:
def to_celsius(f):
    return (f - 32) * 5 / 9

converted_temperature = to_celsius(temps[0])

print(round(converted_temperature, 1))

16.1


In [14]:
movie = {
    "title": "The Matrix",
    "year": 1999,
    "rating": 9.5
}

for key, value in movie.items():
    print(key, ":", value)

title : The Matrix
year : 1999
rating : 9.5


In [15]:
import pandas as pd

In [16]:
chickens = pd.read_csv("chickwts.csv", index_col=0)

In [17]:
chickens.head()
chickens.shape

(71, 2)

In [18]:
chickens.head()

,weight,feed
1,179,horsebean
2,160,horsebean
3,136,horsebean
4,227,horsebean
5,217,horsebean


In [19]:
chickens.shape

(71, 2)

In [20]:
chickens.describe().round(1)

,weight
count,71.0
mean,261.3
std,78.1
min,108.0
25%,204.5
50%,258.0
75%,323.5
max,423.0


## Interpretation of the Dataset

The dataset consists of 71 chickens with an average weight of approximately 261.3. Weights range from 108 to 423, indicating substantial variation across the sample. The median weight is 258, which sits close to the mean and suggests the distribution is roughly symmetric, though a number of birds fall well below or above the typical weight.

## GitHub Repository

https://github.com/raulmckay-blip/cs82a-portfolio